cmd/perf-tests.go

```diff
@@ -422,3 +408,37 @@ func siteNetperf(ctx context.Context, duration time.Duration) madmin.SiteNetPerf
 		TotalConn:       uint64(connectionsPerPeer),
 	}
 }
+
+// perfNetRequest - reader for http.request.body, result should be a pointer if it need
+func perfNetRequest(ctx context.Context, timeout time.Duration, deploymentID string, reqPath string, reader io.Reader, result interface{}) (enpoint string, err error) {
+	cli, err := globalSiteReplicationSys.getAdminClient(ctx, deploymentID)
+	if err != nil {
+		return "", err
+	}
+	rp := cli.GetEndpointURL()
+	reqURL := &url.URL{
+		Scheme: rp.Scheme,
+		Host:   rp.Host,
+		Path:   reqPath,
+	}
+	defer func() {
+		enpoint = rp.String()
```

Comment:
```suggestion
		endpoint = rp.String()
```

Why the defer? Why not just set it and return it as a regular value? 'rp' is not changing from here.

Reply:
because there are four place will write as `rp.String(),nil`. The endpoint should be return always. @klauspost